### PR TITLE
Ensure games have unique game ranks (positions)

### PR DESCRIPTION
**Pull request by gando001**
Display game validation errors when creating a game

### CODE REVIEW

**Comment by gando001** [comment]
1) Validate the game rank positions so that no 2 players can have the same position when creating a game
2) Game validation errors were not displayed (errors were stored in errors[:base] but view used errors[:tournament])

**Comment by proglottis** [comment]
This might be better suited as a setting on each tournament since draws are handled by glicko2 ratings. King of the hill might need to check for this though (Game#reposition_winner)

File: Gemfile [hunk 1]
```diff
@@ -34,5 +34,6 @@ group :test, :development do
   gem 'minitest-spec-rails'
   gem 'capybara_minitest_spec'
   gem 'factory_girl_rails', '~> 4.0'
+  gem 'byebug'
```
Comment (proglottis) [inline Gemfile]: No need to add byebug for this feature

File: app/models/game_rank.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class GameRank < ActiveRecord::Base
 
   has_one :user, :through => :player
 
-  validates_numericality_of :position, greater_than: 0, only_integer: true, if: -> game_rank { game_rank.game.try(:challenged?) }
+  validates_numericality_of :position, greater_than: 0, only_integer: true
```
Comment (proglottis) [inline app/models/game_rank.rb]: Removing the conditional breaks creating challenges and the integration test

**Comment by gando001** [comment]
Useless :)

**Pull request closed without merging**